### PR TITLE
release-24.3: clusterversion: mint 24.3

### PR DESCRIPTION
**Pull request by celiala**
we'll also want to add V24_3 on master (to be done separately)

checklist for next time:

- [x] confirm that any outstanding version gates for blockers have merged
- [x] `./dev test pkg/sql/catalog/bootstrap --rewrite -f TestInitialValuesToString`
- [x] `TestSystemDatabaseSchemaBootstrapVersionBumped`: bump `SystemDatabaseSchemaBootstrapVersion`
- [x] bump version.txt to rc.1

Release note: None
Epic: REL-1307
Release justification: release-process change: mints 24.3.

### CODE REVIEW

**Comment by celiala** [comment]
from [previous run](https://github.com/cockroachdb/cockroach/actions/runs/11707556181/job/32607212054?pr=134342):
- `TestSystemDatabaseSchemaBootstrapVersionBumped`: need to bump `SystemDatabaseSchemaBootstrapVersion`
- `TestUpgradeHappensAfterMigrations`: 
  - current assertion is "require {active version} < {latest version}"
  - but (I think?) assertion should be "require {active version} <= {latest version}"
  - https://github.com/cockroachdb/cockroach/blob/master/pkg/server/migration_test.go#L366

File: pkg/server/migration_test.go [hunk 1]
```diff
@@ -363,7 +363,7 @@ func TestUpgradeHappensAfterMigrations(t *testing.T) {
 					for i := 0; i < N; i++ {
 						runtime.Gosched()
 					}
-					require.True(t, st.Version.ActiveVersion(ctx).Less(clusterversion.Latest.Version()))
+					require.True(t, st.Version.ActiveVersion(ctx).LessEq(clusterversion.Latest.Version()))
```
Comment (celiala) [inline pkg/server/migration_test.go:366]: @RaduBerinde could you confirm that this change is okay to do, i.e.
- current assertion is "require {active version} < {latest version}"
- but (I think?) assertion should be "require {active version} <= {latest version}"
thanks!

**Review by RaduBerinde — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/134342)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @celiala)

___
*[`pkg/clusterversion/cockroach_versions.go` line 320 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/134342#-OB3-NgYAQB6jLxCkUh0:-OB3-NgYAQB6jLxCkUh1:b-nntxhp) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a83b38f10bb69e736e65c7af156fa8e4fa7bb33/pkg/clusterversion/cockroach_versions.go#L320)):*
> ```Go
> 	V24_3_AddTableMetadataCols:                         {Major: 24, Minor: 2, Internal: 22},
> 
> 	// v24.3 versions. Internal versions must be even.
> ```

[nit] remove this comment
___
*[`pkg/server/migration_test.go` line 366 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/134342#gh-1831882403:-OB2zyhA5md5hQgG4zYX:b-vr3kn) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a83b38f10bb69e736e65c7af156fa8e4fa7bb33/pkg/server/migration_test.go#L366)):*
<details><summary><i>Previously, celiala wrote…</i></summary><blockquote>

@RaduBerinde could you confirm that this change is okay to do, i.e.
- current assertion is "require {active version} < {latest version}"
- but (I think?) assertion should be "require {active version} <= {latest version}"
thanks!


</blockquote></details>

Hm.. I'm not too familiar with this but it looks like it was specifically asserting that when `AfterRunPermanentUpgrades` is called, we haven't yet upgraded to the latest (which should be the case because we bootstrap the cluster at `MinSupported`. If you want, I can try to debug it.

**Review by celiala — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/134342)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @RaduBerinde)

___
*[`pkg/clusterversion/cockroach_versions.go` line 320 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/134342#-OB3-NgYAQB6jLxCkUh0:-OB6jkP4ECC_5fjp7rvl:b-dod3l6) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a83b38f10bb69e736e65c7af156fa8e4fa7bb33/pkg/clusterversion/cockroach_versions.go#L320)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

[nit] remove this comment
</blockquote></details>

ah, thanks (done)
___
*[`pkg/server/migration_test.go` line 366 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/134342#gh-1831882403:-OB6ifmj-_OzpqYy9zQ-:byfglse) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a83b38f10bb69e736e65c7af156fa8e4fa7bb33/pkg/server/migration_test.go#L366)):*
> If you want, I can try to debug it.

that would be great -- thank you!

**Comment by RaduBerinde** [comment]
*[`pkg/server/migration_test.go` line 366 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/134342#gh-1831882403:-OB7pQZsCvO5INelHd6c:bccu1cw) ([raw file](https://github.com/cockroachdb/cockroach/blob/1a83b38f10bb69e736e65c7af156fa8e4fa7bb33/pkg/server/migration_test.go#L366)):*
<details><summary><i>Previously, celiala wrote…</i></summary><blockquote>

> If you want, I can try to debug it.

that would be great -- thank you!
</blockquote></details>

I can't reproduce a failure with this PR and switching back to `Less`.. I ran `dev test pkg/server -f TestUpgradeHappensAfterMigrations --stress` and it passed 1000 times.

File: pkg/server/migration_test.go (hunk 1 above)
Comment (celiala) [reply pkg/server/migration_test.go]: okay, this passed for me as well. PR now keeps this test as-is.

(not sure why it failed before -- maybe I didn't update in all the right places, which caused the version to be off in one of the in-progress states?)

anyhow -- appreciate you spending the time digging into it (sorry for the churn)

**Review by celiala — COMMENTED** [review]
Okay, this is ready for review (moving out of draft)

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/134342#-:-OB8DkFw7sy6qpWNHM6I:b-pfyl4u)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @RaduBerinde)

**Review by RaduBerinde — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/134342#-:-OB8brl06eNk0aD0LIrq:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @rail)

**Pull request merged**